internal/backend/remote-state/azure/auth/client_cert_auth.go

```diff
@@ -95,12 +128,21 @@ func (cred *clientCertAuth) AugmentConfig(_ context.Context, config *Config) err
 	return checkNamesForAccessKeyCredentials(config.StorageAddresses)
 }
 
-func decodePFXCertificate(pfxFileName string, password string) (privateKey interface{}, certificate *x509.Certificate, err error) {
-	// read file contents, decode cert
-	contents, err := os.ReadFile(pfxFileName)
+func consolidateCertificate(base64EncodedCertificate, certificateFilename string) ([]byte, error) {
+	certBytes, err := base64.RawStdEncoding.DecodeString(base64EncodedCertificate)
```

Comment:
I'd prefer you to take another pass at this function, and consolidate the error handling above on line 105

Suggestion:
```go
var certBytes []byte
var fileBytes []byte

if len(encodedCert) > 0 {
   certBytes, err = decode(encodedCert)
   if err, return err
}
if len(fileName) > 0 {
  fileBytes, err = read(filename)
  if err, return err
}

hasCert := len(certB) == 0
hasFile := len(fileB) == 0

if hasCert && hasFile {
    // Ensure identical
    return cert
}
if hasCert { return cert }
if hasFile {return file}

return fmt.Errorf("missing")
```